setup.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+# Usage: setup.sh [--help]
+#
+# Environment variables:
+#   VENV: use the specified Python virtual environment. If empty, don't use a
+#     virtual environment and use python/pip from the caller's $PATH. Defaults
+#     to $VIRTUAL_ENV or "$BASEDIR/ocp-venv".
+#   VENV_PYTHON: the python interpreter to use to create the virtual
+#     environment. This only matters, while creating $VENV. If the virtual
+#     environment already exists, or if we are called with VENV="" then this has
+#     no effect. Aside creating the $VENT, we use python/pip from the $PATH/$VENV.
+#   VERBOSE: if set to non-empty, don't hide output of commands.
+#
+# - If you use VENV, you will need to `source "$VENT/bin/activate" before using cda.py.
+# - Sets up bash completion for `oc`. You may want to start a new bash
+#   afterwards or source /etc/bash_completion.d/oc_bash_completion.
+# - If ~/.ssh/id_ed25519 does not exist, a new key gets generated.
+# - The script changes into the top dir of cluster-deployment-automation. All path
+#   names ($VENV) are relative to that path.
+
+set -e
+
+die() {
+    printf '%s\n' "$*"
+    exit 1
+}
+
+usage() {
+    sed -n '3,/^$/ s/^#\( \(.*\)\)\?$/\2/p' "$0"
```

Comment:
Let's try to remove as much as possible from shell scripts and move it into python. The fact that your setup.sh script has a usage string is a red flag.